core/src/test/java/org/htsjdk/core/utils/PathSpecifierUnitTest.java

```diff
@@ -0,0 +1,254 @@
+    public Object[][] getInvalidPaths() {
+        return new Object[][]{
+                // valid URIs that are not valid as a path
+
+                {"file:/project/gvcf-pcr/23232_1#1/1.g.vcf.gz"},    // not encoded
+                {"file://path/to/file.bam"},  // Paths.get throws IllegalArgumentException, 2 leading slashes (vs. 3)
+                // causes "path" to be interpreted as an invalid authority name
+                {"file:project/gvcf-pcr/23232_1#1/1.g.vcf.gz"},     // scheme-specific part is not hierarchichal
+
+                // The hadoop file system provider explicitly throws an NPE if no host is specified and HDFS is not
+                // the default file system
+                //{"hdfs://nonexistent_authority/path/to/file.bam"},  // unknown authority "nonexistent_authority"
+                {"hdfs://userinfo@host:80/path/to/file.bam"},           // UnknownHostException "host"
+
+                // TODO NOTE: the URIs from here down are accepted by IOUtils (public static Path getPath(String uriString))
+                // as valid Paths, even though they are unresolvable and otherwise pretty much useless.
+                {"unknownscheme://foobar"},
+                {"gendb://adb"},
+                {"gcs://abucket/bucket"},
+
+                // URIs with schemes that are backed by an valid NIO provider, but for which the
+                // scheme-specific part is not valid.
+                {"file://nonexistent_authority/path/to/file.bam"},  // unknown authority "nonexistent_authority"
+                {"file://path/to/file.bam"},                        // unknown authority "path"
+        };
+    }
+
+    @Test(dataProvider = "getInvalidPaths")
+    public void testIsPathInvalid(final String invalidPathString) {
+        final PathURI htsURI = new PathSpecifier(invalidPathString);
+        Assert.assertFalse(htsURI.isPath());
+    }
+
+    @Test(dataProvider = "getInvalidPaths", expectedExceptions = {IllegalArgumentException.class, FileSystemNotFoundException.class})
+    public void testToPathInvalid(final String invalidPathString) {
+        final PathURI htsURI = new PathSpecifier(invalidPathString);
+        htsURI.toPath();
+    }
+
+    @DataProvider
+     public Object[][] getInputStreamURIs() throws IOException {
+        return new Object[][]{
+                // URIs that can be resolved to an actual test file
+                {"../data/utils/testTextFile.txt", "Test file."},
+                {"file:///" + getCWD() + "../data/utils/testTextFile.txt", "Test file."},
+
+                // URIs with an embedded fragemnt delimiter ("#"); if the file scheme is included, the rest of the path
+                // must already be escaped; ifno file scheme is included, the path will be escaped by the URI class
+                {"../data/utils/testDirWith#InName/testTextFile.txt", "Test file."},
+                {"file:///" + getCWD() + "../data/utils/testDirWith%23InName/testTextFile.txt", "Test file."},
```

Comment:
Need to resolve how to reference these paths without "..".

Reply:
I think that you can use `Paths.get(getCWD() + "../data/utils/testDirWith%23InName/testTextFile.txt").toUri().toString()`, if the problem is with Windows. I can test if you want locally and let you know.

Reply:
There are several problems here; the original one about where/how we should store test data, and the Windows one where the path separator is different. The latter has two parts 1) how we specify paths in tests 2) How we test Windows paths (UNC paths, dos drive specifiers, encoded names with embedded spaces etc).

Also, I know you're not yet convinced we need these classes, but for these tests in particular, I want to make sure the raw strings are passed directly to the classes being tested, not strings that have been pre-processed by `Paths` and `URI`, since I'm trying to test that the classes themselves do this correctly.

Reply:
Also, there is the issue that Windows paths (with forward slashes) don't appear to be valid URIs ?